gio/Gir.toml

```diff
@@ -1187,6 +1225,20 @@ status = "generate"
         [object.function.return]
         nullable = true
 
+[[object]]
+name = "Gio.SocketControlMessage"
+status = "generate"
+manual_traits = ["SocketControlMessageExtManual"]
+    [[object.function]]
+    name = "deserialize"
+    # writes from a void pointer
+    manual = true
```

Comment:
Should probably just get `element-type` annotations in C?

Reply:
Seems to have it already: [gsocketcontrolmessage.c#L157](https://gitlab.gnome.org/GNOME/glib/-/blob/96bc6d6e7cdc1cd24e54f1499092c0354991e391/gio/gsocketcontrolmessage.c#L157)

Reply:
Hm, can you create a gir issue?

Reply:
Not sure why it was screwing up before but it works now, removed the manual implementation

Reply:
Oh now I remember, it detects the right type but does not cast the pointer correctly to `gpointer`

Reply:
https://github.com/gtk-rs/gir/pull/1407

Reply:
Merged now, can you regenerate? I think then we're ready here? :)